Replicator/Replicator.cc

```diff
@@ -1126,11 +1128,18 @@ namespace litecore { namespace repl {
             return;
         }
         
-        unordered_set<C4Database::CollectionSpec> specs;
-        for (auto &coll : _collections) {
-            specs.insert(coll->getSpec());
+        unordered_map<C4Database::CollectionSpec, C4Collection*> passiveCollections;
+        for (int i = 0; i < _collections.size(); i++) {
+            if (_options->pushOf(i) == kC4Passive || _options->pullOf(i) == kC4Passive) {
```

Comment:
I still refuse to believe that we mix active and passive.  This should be an assertion, not a check that makes the code less readable and guards in multiple places against an invalid situation.

Reply:
From slack discussion, there shouldn't be mix and match b/w active and passive collections. The assertion will be done when the Replicator is created. The filter will be remove from here.